### PR TITLE
Remove unused perfcounter code

### PR DESCRIPTION
**Pull request by sywhang**
Perf counters are not enabled on CoreCLR and we're enabling a different way to deliver a similar experience on cross-platform (see #20372).

This PR removes the following dead code:
- ```COUNTER_ONLY``` macro
- ```perfcounter*``` files under ```src\inc\```
- ```ENABLE_PERF_COUNTERS``` macro and any code that lived under ```#ifdef ENABLE_PERF_COUNTERS```.
- ```PerfCountersSupportedBuild```

I didn't touch the GC side on purpose.

### CODE REVIEW

**Comment by vancem** [comment]
@AaronRobinsonMSFT  - I have been working closely with @sywhang  and @noahfalk  on the replacement for PerfCounters (called EventCounters) see https://github.com/dotnet/coreclr/issues/20372 for details.   Fundamentally, PerfCounters were problematic because they were windows specific, and repesented a DIFFERENT pipe for telemetry then we use for Events, which is why they are not being pursued in .NET Core.   @sywhang is plumbing in the replacement right now, so getting rid of now abandoned code is good.   We don't need breadcrumbs.

File: src/gc/gcee.cpp [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
 #ifndef DACCESS_COMPILE
 
+#define COUNTER_ONLY(x)
```
Comment (jkotas) [inline src/gc/gcee.cpp]: Could you please make this:

```
#if defined(ENABLE_PERF_COUNTERS)
#define COUNTER_ONLY(x)
#endif
```

Comment (jkotas) [reply src/gc/gcee.cpp]: Actually, all places where this macro is used in this file are inside `ENABLE_PERF_COUNTERS` already, except the first one. So I would just change the following two lines to: 

```
#ifdef ENABLE_PERF_COUNTERS
uint64_t g_TotalTimeInGC = 0;
uint64_t g_TotalTimeSinceLastGCEnd = 0;
#endif // ENABLE_PERF_COUNTERS
```

and delete this macro from this file.

File: src/gc/handletable.cpp [hunk 2]
```diff
@@ -21,6 +21,8 @@
 #include "objecthandle.h"
 #include "handletablepriv.h"
 
+#define COUNTER_ONLY(x)
```
Comment (jkotas) [inline src/gc/handletable.cpp]: Just change the single place where this is used to `#ifdef ENABLE_PERF_COUNTERS`

Comment (jkotas) [reply src/gc/handletable.cpp]: And delete `src\gc\env\gcenv.base.h(501): #define COUNTER_ONLY(x)` to make the `COUNTER_ONLY` macro to disappear completely. We are just going to have `ENABLE_PERF_COUNTERS` in the gc.

**Pull request merged**